### ISSUE TITLE
[Feature request] Add predefined regions from python's regionmask

### ISSUE THREAD

**Issue by briochemc** [natgeo-wong/GeoRegions.jl#5]
I mentioned this on Slack a little while ago, but I thought I'd open an issue for your record, in case you have time to look into it. 

It would be great to add the regions that python's [regionmask](https://regionmask.readthedocs.io/en/stable/index.html) currently offers on top of scientific regions:
- [ ] [countries / states](https://regionmask.readthedocs.io/en/stable/defined_countries.html)
- [ ] [land masks regions](https://regionmask.readthedocs.io/en/stable/defined_landmask.html)
- [ ] and [marine / ocean regions](https://regionmask.readthedocs.io/en/stable/defined_ocean_basins.html)

I'm also pinging @asinghvi17 since I stumbled upon his related (but empty!) [NaturalEarth.jl](https://github.com/asinghvi17/NaturalEarth.jl) repository, since all of the regions listed above seem to come from [naturalearthdata.com](https://www.naturalearthdata.com/)!

**Comment by natgeo-wong** [comment]
Hi @briochemc, do you have the shapefiles, or some form of raw data that I can have a look at to see how best to extract and put into the package?  I definitely would be interested in putting all the regions above into GeoRegions.jl

**Comment by briochemc** [comment]
No but they are downloaded automatically via cartopy in the python regionmask tool, so I think you can probably translate ideas from https://github.com/regionmask/regionmask/blob/stable/regionmask/defined_regions/natural_earth.py, no?

**Comment by natgeo-wong** [comment]
That probably means that I'll need to download the files directly from natural earth and explore them.  I can do that.

Alternatively, I could have a look and see if there is output from regionmask in the lon/lat form and then export it to a textfile or something.  But thanks or bringing this up - need some documentation of what to do.

**Comment by natgeo-wong** [comment]
I would also add that a global landmask isn't really possible with the functionality of GeoRegions since this would require the combination of multiple GeoRegions that are not otherwise spatially connected.

What would be more appropriate for a landmask would be do download a land-sea mask and apply a filter based on land-sea mask values onto the data.

**Comment by briochemc** [comment]
Yep, I had to do some manual stitching for OceanBasins myself :) 

An ideal solution would be to have polygon/region shapes defined on the sphere but I don't know where to look for that directly 🤷 . Meshes.jl maybe?

**Comment by natgeo-wong** [comment]
@briochemc if you are interested I might getting around to tackle adding the shapefiles from OceanBasins.jl to GeoRegions.jl myself

**Comment by briochemc** [comment]
Yes, please do!

**Comment by briochemc** [comment]
Did you end up doing this in the end?

**Comment by natgeo-wong** [comment]
Oops not yet, I'm kinda swamped with other things and rewriting GeoRegions.jl (again). Many apologies. I'll get to it.

**Comment by asinghvi17** [comment]
You can probably just get the regions from NaturalEarth.jl if you're ok with downloading on the fly!

**Comment by briochemc** [comment]
No need to apologize! 😅

**Comment by natgeo-wong** [comment]
Hey @briochemc, I want to let you know that I have had a look at the OceanBasins dataset from NaturalEarth. At this time, because there are several regions in the OceanBasins dataset which comprise **multiple** polygons, GeoRegions.jl is not really compatible with this dataset, because each GeoRegion is a singular polygon.

However, in v8 of GeoRegions.jl, I intend to add GeoRegions type for regions which are defined by multiple Polygonal shapes, so hopefully this feature will not be too far off in the future (within the next few months, hopefully). I just need to get RegionGrids.jl and LandSea.jl off the ground first.
